pyop2/types/dat.py

```diff
@@ -722,39 +723,11 @@ def halo_valid(self, value):
 
     @property
     def data(self):
-        full = self._parent.data
-        idx = (slice(None), *self.index)
-        return full[idx]
+        return self._parent.data[self._idx]
 
     @property
-    def data_ro(self):
-        full = self._parent.data_ro
-        idx = (slice(None), *self.index)
-        return full[idx]
-
-    @property
-    def data_wo(self):
-        full = self._parent.data_wo
-        idx = (slice(None), *self.index)
-        return full[idx]
-
-    @property
-    def data_with_halos(self):
-        full = self._parent.data_with_halos
-        idx = (slice(None), *self.index)
-        return full[idx]
-
-    @property
-    def data_ro_with_halos(self):
-        full = self._parent.data_ro_with_halos
-        idx = (slice(None), *self.index)
-        return full[idx]
-
-    @property
-    def data_wo_with_halos(self):
-        full = self._parent.data_wo_with_halos
-        idx = (slice(None), *self.index)
-        return full[idx]
+    def _data(self):
+        return self._parent._data[self._idx]
```

Comment:
Will halo updates still do the right thing after this change and mark the parent's halos as dirty?

Reply:
It should, as the halo_valid property is mapped to that of the parent https://github.com/OP2/PyOP2/blob/5316660eb1ef9a84a1484293bb23083a85b2b60d/pyop2/types/dat.py#L716-L722